pyxem/generators/variance_generator.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2018 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Variance generators in real and reciprocal space for fluctuation electron
+microscopy.
+
+"""
+
+import numpy as np
+from hyperspy.signals import Signal2D
+from hyperspy.api import stack
+
+from pyxem.signals.diffraction_variance import DiffractionVariance
+from pyxem.signals.diffraction_variance import ImageVariance
+
+
+class VarianceGenerator():
+    """Generates variance images for a specified signal and set of aperture
+    positions.
+
```

Comment:
This is not what this class currently does - there is no way to specify aperture positions at present. It could be possible to pass a VDFGenerator to the VarianceGenerator to make that functionality happen (may be interesting) but more importantly the docstring needs to match the code.